From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <jeremya@chromium.org>
Date: Thu, 20 Sep 2018 17:50:38 -0700
Subject: leveldb_ssize_t.patch

Fix conflict between leveldb & node's definition of ssize_t on
Windows by preventing leveldb from re-defining the type if it's
already defined.

diff --git a/third_party/leveldatabase/port/port_chromium.h b/third_party/leveldatabase/port/port_chromium.h
index a7c449eba19c3e3f46eec9c428fb497f6b1a7852..acbce7efd582e226419054cf80e1c27f4919b2fb 100644
--- a/third_party/leveldatabase/port/port_chromium.h
+++ b/third_party/leveldatabase/port/port_chromium.h
@@ -26,7 +26,11 @@
 #endif
 
 #if defined(OS_WIN)
+#  if !defined(_SSIZE_T_) && !defined(_SSIZE_T_DEFINED)
 typedef SSIZE_T ssize_t;
+#  define _SSIZE_T_
+#  define _SSIZE_T_DEFINED
+#  endif
 #endif
 
 namespace leveldb {
